cw/tvla.py

```diff
@@ -865,11 +856,15 @@ def run_tvla(ctx: typer.Context):
 
             log.info("Plotting T-test Statistics vs. Number of Traces, this may take a while.")
             # Determine resolution.
-            xres_vec = [10e6, 1e6, 100e3, 10e3, 1e3]
+            xres_vec = [10e6, 1e6, 100e3, 10e3, 1e3, 100, 1]
             for xres in xres_vec:
                 if int(np.around(trace_end_vec[0] / xres)) > 0:
-                    xres_label = str(int(xres / 1e6)) + 'M' if xres >= 1e6 else \
-                        str(int(xres / 1e3)) + 'k'
+                    if xres >= 1e6:
```

Comment:
It seems that we have a pre-existing problem in the `if` condition in line 861. It would be appropriate to fix it with this commit.

`trace_end_vec[0] ` should be changed to `(trace_end_vec[0] - trace_start_vec[0])`

E.g. Currently if I run TVLA on traces 4000-5000 with `number_of_steps: 10`, the x-axis is indexed in multiples of `1k ` when it should be in multiples of `100`.

Reply:
Uh, very good catch @vrozic ! Let me change that.

Reply:
Done.